File: src/python/thread-1.py
# thread - criar thread e empilhar para execução
# 

# Importa as libs padrões
import threading
import time

# função que será executada na thread
def minha_funcao():
    print("Thread iniciada!")
    time.sleep(2)
    print("Thread finalizada!")

def main():
    # Criar a thread
    thread = threading.Thread(target=minha_funcao)

    # Iniciar a thread
    thread.start()

    # Aguardar a thread terminar
    thread.join()
    print("Programa principal finalizado!")

if __name__ == "__main__":
    main()

File: src/go/thread-1.go
package main

import (
    "fmt"       
    "sync"      
    "time"      
)

func minhaFuncao(wg *sync.WaitGroup) {
    defer wg.Done() 
    fmt.Println("Thread (goroutine) iniciada!")
    time.Sleep(2 * time.Second) 
    fmt.Println("Thread (goroutine) finalizada!")
}

func main() {
    var wg sync.WaitGroup 

    wg.Add(1)             
    go minhaFuncao(&wg)  

    wg.Wait()             
    fmt.Println("Programa principal finalizado!")
}
